onnxruntime/core/common/threadpool.cc

```diff
@@ -26,6 +26,82 @@ namespace onnxruntime {
 
 namespace concurrency {
 
+ThreadPoolProfiler::ThreadPoolProfiler() {
+  memset(events_, 0, sizeof(uint64_t) * MAX_EVENT);
+}
+
+bool ThreadPoolProfiler::Enabled() const {
+  return enabled_ && std::this_thread::get_id() == main_thread_id_;
+}
+void ThreadPoolProfiler::Start() {
+  enabled_ = true;
```

Comment:
should we check that we are not already started? otherwise perhaps main_thread_id_ can be overwritten.

Reply:
Sure we could add an assert.